tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -124,6 +125,33 @@ var _ = Describe("odo devfile create command tests", func() {
 			values := gjson.GetMany(output, "kind", "metadata.name", "status.state")
 			Expect(helper.GjsonMatcher(values, []string{"Component", "nodejs", "Unknown"})).To(Equal(true))
 		})
+
+		It("should successfully create the devfile component and show json output for a unreachable cluster", func() {
+
+			path := os.Getenv("KUBECONFIG")
+
+			// read the contents from the kubeconfig and replace the server entries
+			reg := regexp.MustCompile(`server: .*`)
+			kubeConfigContents, err := helper.ReadFile(path)
+			Expect(err).To(BeNil())
+			kubeConfigContents = reg.ReplaceAllString(kubeConfigContents, "server: https://not-reachable.com:443")
+
+			// write to a new file which will be used as the new kubeconfig
+			newKubeConfigPath := filepath.Join(commonVar.Context, "newKUBECONFIG")
+			newKubeConfig, err := os.Create(newKubeConfigPath)
+			Expect(err).To(BeNil())
+			_, err = newKubeConfig.WriteString(kubeConfigContents)
+			Expect(err).To(BeNil())
+
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(devfilePath))
```

Comment:
Why do we copy the devfile here? Wouldn't running the cmd below would create a devfile?

Reply:
This saves a network call to the registry and instead uses the provided devfile.